internal/system/agent/router.go

```diff
@@ -37,43 +50,53 @@ func LoadRestRoutes() *mux.Router {
 	return r
 }
 
-func operationServiceHandler(w http.ResponseWriter, r *http.Request) {
+func operationHandler(w http.ResponseWriter, r *http.Request) {
 	if r.Body != nil {
 		defer r.Body.Close()
 	}
 
 	// TODO: Work with parsing mux.Vars(r) and assigning to vars.
-	//vars := mux.Vars(r)
-	//action := vars["ops"]
-	//params :vars["params"]
-	//services := vars["services"]
-	var action string
-	var params map[string]string
 	var services []string
+	vars := mux.Vars(r)
+	action := vars["action"]
+
+	var params []string
+	var o models.Operation
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&o)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error decoding operation: " + err.Error())
+		return
+	}
 
-	switch action {
+	action = o.Action
+	services = o.Services
+	params = o.Params
 
-	// Make asynchronous call(s) to the appropriate internal function (to stop, start, or restart the service(s).
+	switch action {
 
-	case START:
-		go invokeAction(START, services, params)
-		w.WriteHeader(http.StatusCreated)
-		w.Write([]byte("Response"))
+	// Make asynchronous goroutine call(s) to the appropriate internal function (respectively, to stop, start, or restart the service(s).
+	case STOP:
+		go InvokeOperation(STOP, services, params)
+		w.WriteHeader(http.StatusOK)
```

Comment:
Without some kind of result from the `go InvokeOperation` how do you know that the operation was successful and we should return a StatusOK? What if the user hands the service an invalid service key?